.github/workflows/ci-lint-test.yml

```diff
@@ -6,29 +6,80 @@ name: Lint Checks
 on:
   push:
     branches: [main]
-
   pull_request:
     branches: [main]
 
 concurrency:
   group: ${{ github.workflow }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}
   cancel-in-progress: true
 
-# See https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions
-permissions:  # added using https://github.com/step-security/secure-workflows
+permissions:
   contents: read
 
 jobs:
-  generated-files-check:
+  setup-scripts:
     runs-on: ubuntu-latest
     steps:
-      - uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
+      - uses: step-security/harden-runner@v2
         with:
           egress-policy: audit
+      - name: Clone Jaeger main repo scripts
+        run: |
+          git clone --depth 1 https://github.com/jaegertracing/jaeger.git /tmp/jaeger
+          cp -r /tmp/jaeger/scripts ./scripts
 
-      - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+  lint:
+    needs: setup-scripts
```

Comment:
needs: means this job will run after setup-scripts, but they do not share the file system

Reply:
how is this resolved? It's not going to work.

Reply:
?